src/openforms/js/components/admin/form_design/logic/OperandTypeSelection.js

```diff
@@ -8,33 +8,19 @@ import Select from '../../forms/Select';
 const OPERAND_TYPES = {
     literal: defineMessage({description: '"literal" operand type', defaultMessage: 'value'}),
     component: defineMessage({description: '"component" operand type', defaultMessage: 'the field'}),
-    today: defineMessage({description: '"today" operand type', defaultMessage: 'today'})
-};
-
-const COMPONENT_TYPE_TO_OPERAND_TYPE = {
-    number: {
-        literal: OPERAND_TYPES.literal,
-        component: OPERAND_TYPES.component,
-    },
-    textfield: {
-        literal: OPERAND_TYPES.literal,
-        component: OPERAND_TYPES.component,
-    },
-    iban: {
-        literal: OPERAND_TYPES.literal,
-        component: OPERAND_TYPES.component,
-    },
-    date: {...OPERAND_TYPES},
+    today: defineMessage({description: '"today" operand type', defaultMessage: 'today'}),
+    array: defineMessage({description: '"array" operand type', defaultMessage: 'the array'}),
 };
 
 
-const OperandTypeSelection = ({name, operandType, componentType, onChange}) => {
+const OperandTypeSelection = ({name, operandType, onChange, filter}) => {
     const intl = useIntl();
-    const choices = Object.entries(COMPONENT_TYPE_TO_OPERAND_TYPE[componentType]);
+    const choices = getTranslatedChoices(intl, OPERAND_TYPES);
+
     return (
         <Select
             name={name}
-            choices={getTranslatedChoices(intl, choices)}
+            choices={filter ? choices.filter(filter) : choices}
```

Comment:
I'd specify a default value for `filter` which is `allowAny` (`const allowAny = () => true;`) so that you can _always_ call the `filter` function provided.

Usually it's best practice to provide a default value for props that are not required.